pennylane/_device.py

```diff
@@ -178,13 +178,45 @@ def shots(self, shots):
         Raises:
             DeviceError: if number of shots is less than 1
         """
-        if shots < 1:
+        if isinstance(shots, int):
+            if shots < 1:
+                raise DeviceError(
+                    "The specified number of shots needs to be at least 1. Got {}.".format(shots)
+                )
+
+            self._shots = int(shots)
+            self._shot_vector = None
+
+        elif isinstance(shots, Sequence):
+            # a sequence of shots has been provided
+
+            if all(isinstance(s, int) for s in shots):
```

Comment:
I wonder if we could drop this check, since it iterates through the entire list? I know that this will only happen once at device creation, but if users provide wrong inputs, the error should be thrown where it breaks (as long as one can legibly  figure out what was wrong), no?

Reply:
I wasn't sure how much validation to do here 🤔 

For `shots=n`, we check not only if `n` is an integer, but if `n>1`.